solr/core/src/java/org/apache/solr/core/SolrResourceLoader.java

```diff
@@ -915,27 +921,22 @@ private PackageListeningClassLoader createSchemaLoader() {
 
   public static void persistConfLocally(SolrResourceLoader loader, String resourceName, byte[] content) {
     // Persist locally
-    File confFile = new File(loader.getConfigDir(), resourceName);
+    Path confFile = loader.getConfigPath().resolve(resourceName);
     try {
-      File parentDir = confFile.getParentFile();
-      if (!parentDir.isDirectory()) {
-        if (!parentDir.mkdirs()) {
-          final String msg = "Can't create managed schema directory " + parentDir.getAbsolutePath();
-          log.error(msg);
-          throw new SolrException(SolrException.ErrorCode.SERVER_ERROR, msg);
-        }
-      }
-      try (OutputStream out = new FileOutputStream(confFile);) {
-        out.write(content);
+      Path parentDir = confFile.getParent();
+      if (!Files.isDirectory(parentDir)) {
+        Files.createDirectories(parentDir);
```

Comment:
according to javadocs, createDirectories doesn't throw if parentDir already exists so you needn't conditionally check.